src/OracleService/OracleService.cs

```diff
@@ -0,0 +1,488 @@
+            if (FinishedCache.ContainsKey(requestId)) throw new RpcException(-100, "Request has already finished");
+
+            using (SnapshotView snapshot = Blockchain.Singleton.GetSnapshot())
+            {
+                if (NativeContract.Oracle.GetRequest(snapshot, requestId) is null)
+                    throw new RpcException(-100, "Request is not found");
+                AddResponseTxSign(snapshot, requestId, oraclePub, txSign);
+            }
+            return new JObject();
+        }
+
+        [ConsoleCommand("start oracle", Category = "Oracle", Description = "Start oracle service")]
+        private void OnStart()
+        {
+            string password = ConsoleBase.ReadUserInput("password", true);
+            if (password.Length == 0)
+            {
+                Console.WriteLine("Cancelled");
+                return;
+            }
+            try
+            {
+                Wallet.Unlock(password);
+            }
+            catch (System.Security.Cryptography.CryptographicException)
+            {
+                Console.WriteLine($"Failed to open wallet");
+                return;
+            }
+
+            using (var snapshot = Blockchain.Singleton.GetSnapshot())
+            {
+                if (!CheckOracleAvaiblable(snapshot, out ECPoint[] oracles)) throw new ArgumentException("The oracle service is unavailable");
+                if (!CheckOracleAccount(oracles)) throw new ArgumentException("There is no oracle account in wallet");
+            }
+            Interlocked.Exchange(ref CancelSource, new CancellationTokenSource())?.Cancel();
+            new Task(() =>
+            {
+                while (CancelSource?.IsCancellationRequested == false)
+                {
+                    using (var snapshot = Blockchain.Singleton.GetSnapshot())
+                    {
+                        IEnumerator<(ulong RequestId, OracleRequest Request)> enumerator = NativeContract.Oracle.GetRequests(snapshot).GetEnumerator();
+                        while (enumerator.MoveNext() && !CancelSource.IsCancellationRequested)
+                            if (!FinishedCache.ContainsKey(enumerator.Current.RequestId) && (!PendingQueue.TryGetValue(enumerator.Current.RequestId, out OracleTask task) || task.Tx is null))
+                                ProcessRequest(snapshot, enumerator.Current.RequestId, enumerator.Current.Request);
+                    }
+                    Thread.Sleep(500);
+                }
+            }).Start();
+
+            if (Timer is null)
+            {
+                Timer = new System.Timers.Timer();
+                Timer.Enabled = true;
+                Timer.Interval = RefreshInterval;
+                Timer.Start();
+                Timer.Elapsed += new ElapsedEventHandler(OnTimer);
+            }
+        }
+
+        void OnPersist(StoreView snapshot, IReadOnlyList<Blockchain.ApplicationExecuted> applicationExecutedList)
```

Comment:
Where is this method used?

Reply:
But no one calls it.

Reply:
It's the interface of `IPersistencePlugin`, which will be called by neo-core.

Reply:
```suggestion
        void IPersistencePlugin.OnPersist(StoreView snapshot, IReadOnlyList<Blockchain.ApplicationExecuted> applicationExecutedList)
```

Reply:
If you don't add `IPersistencePlugin.`, it won't be called.

Reply:
Thanks, fixed.